controller.go

```diff
@@ -337,6 +341,14 @@ func (c *controller) clusterAgentInit() {
 			// service bindings
 			c.agentClose()
 			c.cleanupServiceBindings("")
+
+			c.Lock()
+			if c.agentStopDone != nil {
+				close(c.agentStopDone)
+				c.agentStopDone = nil
```

Comment:
How does `AgentStopWait` work if the channel is set to nil? Is the external code expected to call `AgentStopWait` before triggering the shutdown?